src/js/actions/documents.js

```diff
@@ -273,27 +273,36 @@ define(function (require, exports) {
      * Initialize document and layer state, emitting DOCUMENT_UPDATED events, for
      * all the inactive documents.
      *
-     * @param {number} currentIndex
+     * @param {number} currentDocID
      * @param {number} docCount
      * @return {Promise}
      */
-    var initInactiveDocuments = function (currentIndex, docCount) {
-        var otherDocPromises = _.range(1, docCount + 1)
-            .filter(function (index) {
-                return index !== currentIndex;
-            })
+    var initInactiveDocuments = function (currentDocID, docCount) {
+        var docReferences = _.range(1, docCount + 1)
             .map(function (index) {
-                var indexRef = documentLib.referenceBy.index(index);
-
-                // Only load essential properties for inactive documents
-                return _getInactiveDocumentByRef(indexRef)
-                    .bind(this)
-                    .then(function (document) {
-                        this.dispatch(events.document.DOCUMENT_UPDATED, {
-                            document: document
-                        });
+                return documentLib.referenceBy.index(index);
+            }),
+            otherDocPromises = descriptor.batchGetProperty(docReferences, "documentID")
+                .then(function (documentIDs) {
+                    return documentIDs.filter(function (documentID) {
+                        return documentID !== currentDocID;
                     });
-            }, this);
+                })
+                .bind(this)
```

Comment:
For consistency, move the `.bind(this)` up to being the first method called on the original promise, after the `descriptor` call.